### ISSUE TITLE
mkdir issue after switching to ./run_1940_cluster.sh in start_das.sh

### ISSUE THREAD

**Issue by aflaxman** [p-b-j/census-das-container#4]
I'm following the [CLUSTER.md](https://github.com/p-b-j/census-das-container/blob/main/CLUSTER.md#running-an-applcation) instructions and I think I've got a spark clustering running on top of SGE. Awesome!  I'm hitting a mkdir error, though, which looks like this:
```
(base) [abie@long-uge-archive-p126 census-das-container]$ time singularity run --home /share/scratch/users/abie/ --bind singularity_tmp:/tmp census_das.img
...
py.test tests/gurobi_2_test.py
============================================================== test session starts ==============================================================
platform linux -- Python 3.6.5, pytest-3.5.1, py-1.5.3, pluggy-0.6.0
rootdir: /ihme/homes/abie/projects/2021/census-das-container/census2020-das-e2e, inifile:
plugins: remotedata-0.2.1, openfiles-0.3.0, doctestplus-0.1.3, arraydiff-0.2
collected 1 item

tests/gurobi_2_test.py .                                                                                                                  [100%]

=========================================================== 1 passed in 10.52 seconds ===========================================================
mkdir: cannot create directory ▒/home/abie▒: Read-only file system
```

**Comment by p-b-j** [comment]
Awesome the spark cluster is running! This is the same error from #2 but of course I somehow didn't copy the fix over when creating the cluster script... something like "fool me once shame on them, fool me twice shame on me" echoing in my head right now.

Anyways, I just pushed the fix which I think should resolve that issue, let me know when you have a chance to try again!

**Comment by aflaxman** [comment]
Your fix got me past that error, but now I have an issue with the path for python:
```
  File "/usr/local/spark/spark-2.4.7-bin-hadoop2.7/python/lib/py4j-0.10.7-src.zip/py4j/protocol.py", line 328, in get_return_value
py4j.protocol.Py4JJavaError: An error occurred while calling z:org.apache.spark.api.python.PythonRDD.runJob.
: org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 27, 10.158.96.148, executor 3): java.io.IOException: Cannot run program "/usr/local/anaconda3/bin/python3.6": error=2, No such file or directory
```
I have lots of options:
```
(base) [abie@long-uge-archive-p126 census-das-container]$ whereis python3.6
python3: /opt/infra/bin/python3.8-config /opt/infra/bin/python3.8 /opt/infra/bin/python3 /opt/set_uge_cgroup/bin/python3 /opt/salt/bin/python3 /opt/kill_orphans/bin/python3 /ihme/homes/abie/miniconda3/bin/python3 /ihme/homes/abie/miniconda3/bin/python3.7 /ihme/homes/abie/miniconda3/bin/python3.7-config /ihme/homes/abie/miniconda3/bin/python3.7m /ihme/homes/abie/miniconda3/bin/python3.7m-config
```

**Comment by p-b-j** [comment]
Ah I think this is where the interactions get weird between the application and the workers since the app is in a singularity container/environment and the workers are not. I'll do some research/reading, but my initial thought is that the best route to go down is to have everything running in a container. And ideally setting this up in a way that allows us to reuse the census_das.img (I think singularity has some options for changing the entrypoint script) since we really only need to change the command that is run when we start the container.

Long story short, I'll update here when I think I have a fix!

**Comment by p-b-j** [comment]
Ok updated https://github.com/p-b-j/census-das-container/blob/main/CLUSTER.md with new instructions for setting up a cluster and running DAS. I was able to setup a cluster following these instructions on my local machine and successfully submit the DAS, but I'm not sure if that success will translate to the SGE.

One thing I'll note is the cluster only made progress when I explicitly set the memory/cores when starting the worker nodes. I think this was due to the weirdness of running all of the nodes on the same machine, but it might be something to be aware of.

**Comment by p-b-j** [comment]
Finally got around to making the commands much nicer to run since our last conversation. The repo structure shifted a bit, but the new instructions are viewable in [cluster/README.md](https://github.com/p-b-j/census-das-container/blob/main/cluster/README.md).

I changed some of the cluster setup instructions as well, so it might be good to go through that entire README to make sure everything is setup properly!

**Comment by aflaxman** [comment]
Seems clear and working.  Now let’s see if it runs to completion!

**Comment by aflaxman** [comment]
Successful completion in about 3 hours. :thumbs up emoji:

From: Abraham D Flaxman <abie@uw.edu>
Sent: Monday, March 8, 2021 1:22 PM
To: p-b-j/census-das-container <reply@reply.github.com>; p-b-j/census-das-container <census-das-container@noreply.github.com>
Cc: Abraham Flaxman <abie@alum.mit.edu>; Author <author@noreply.github.com>
Subject: RE: [p-b-j/census-das-container] mkdir issue after switching to ./run_1940_cluster.sh in start_das.sh (#4)

Seems clear and working.  Now let’s see if it runs to completion!

**Comment by p-b-j** [comment]
Awesome glad to hear it! Closing this issue for now, it'd be cool to see how much speed up we can get w/different worker parameters but we can talk more about this later
